pkg/master/master_test.go

```diff
@@ -118,6 +121,20 @@ func setUp(t *testing.T) (*etcd3testing.EtcdTestServer, Config, *assert.Assertio
 	}
 	config.ExtraConfig.VersionedInformers = informers.NewSharedInformerFactory(clientset, config.GenericConfig.LoopbackClientConfig.Timeout)
 
+	c, err := config.ExtraConfig.StorageFactory.NewConfig(api.Resource("apiServerIPInfo"))
```

Comment:
this doesn't belong in the setup method. make a specific TestEmptySubsets test that calls setUp, then creates an endpoints object with the empty subsets, then calls ListLeases()

Reply:
you should test masterCountEndpointReconciler's RemoveEndpoints() and storageLeases's ListLeases() in controller_test.go and reconcilers/lease_test.go

Reply:
I stepped through store#List and tried to see how Endpoints object with empty subsets can be injected so that ListLeases() iterates over the Items.

Reply:
Looking at masterCountEndpointReconciler, it doesn't have field of Leases type.

Reply:
Personal perspective, changes in pkg/master/reconcilers/mastercount.go can be tested like here 
https://github.com/kubernetes/kubernetes/blob/0155d18fbc798031fe904bc99abe2b935058beb8/pkg/master/controller_test.go#L397
rather than this.